### DISCUSSION TITLE
A scrollable list of items with too many elements

### DISCUSSION THREAD

**Discussion by sairus7** [ocornut/imgui#3954] [Old/Archived discussions]
I want to show an array of 100000 strings into a listbox with single item selection and scrolling.
But it is very inefficient to render them all every frame.
So, I should somehow:
- Draw only current visible items:
  - get number of elements that can fit into the current listbox height
  - get first and last index of visible elements from scrollbar position and draw only them
- Draw a cusom scrollbar, since I am filling listbox only with few items and built-it scrollbar would not show properly:
  - add VSlider at the same line
  - revert its min-max values, and recalc first and last visible index

Are there some ready-to use widgets with these optimizations enabled?
Or maybe I can modify widget scrollbar behaviour without adding vertical slider?

**Comment by ocornut** [comment]
This problem is already solved by ImGuiListClipper and demonstrated in multiple places of the demo.

**Reply by sairus7** [reply]
Thanks! You saved my day!
